waiter/src/waiter/util/http_utils.clj

```diff
@@ -45,15 +45,16 @@
         {:keys [error status] :as response} (async/<!! raw-response)]
     (when error
       (throw error))
-    (when (and throw-exceptions (not (<= 200 status 299)))
-      (ss/throw+ response))
-    (let [response-body (-> response :body async/<!!)]
-      (try
-        (cond-> response-body
-          (not-empty response-body)
-          (-> json/read-str walk/keywordize-keys))
-        (catch Exception _
-          response-body)))))
+    (let [response (update response :body async/<!!)]
+      (when (and throw-exceptions (not (<= 200 status 299)))
+        (ss/throw+ response))
+      (let [{:keys [body]} response]
+        (try
+          (cond-> body
+            (not-empty body)
+            (-> json/read-str walk/keywordize-keys))
+          (catch Exception _
+            body))))))
```

Comment:
Would it be possible to write an integration test for this?

Reply:
Added a unit test :)